test/batchReceiver.spec.ts

```diff
@@ -421,4 +423,533 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].deliveryCount, 0);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await testPeekMsgsLength(queueClient, 1);
+
+    receivedMsgs = await queueClient.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].deliveryCount, 1);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+  });
+
+  it("No settlement of the message is retained in the Subscriptions with incremented deliveryCount", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    let receivedMsgs = await subscriptionClient.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await testPeekMsgsLength(subscriptionClient, 1);
+
+    receivedMsgs = await subscriptionClient.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+  });
+
+  it("Receive n messages but queue only has m messages, where m < n", async function(): Promise<
+    void
+  > {
+    await queueClient.send(testMessages[0]);
+    const receivedMsgs = await queueClient.receiveBatch(2);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+
+    await testPeekMsgsLength(queueClient, 0);
+  });
+
+  it("Receive n messages but subscription only has m messages, where m < n", async function(): Promise<
+    void
+  > {
+    await topicClient.send(testMessages[0]);
+    const receivedMsgs = await subscriptionClient.receiveBatch(2);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receivedMsgs[0].complete();
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  async function deferMsgAndGetSequenceNum(
+    client: QueueClient | SubscriptionClient
+  ): Promise<Long> {
+    const receivedMsgs = await client.receiveBatch(1);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    if (!receivedMsgs[0].sequenceNumber) {
+      throw "Sequence Number can not be null";
+    }
+    const sequenceNumber = receivedMsgs[0].sequenceNumber;
+    await receivedMsgs[0].defer();
+    return sequenceNumber;
+  }
+
+  it("Abandon a message received from recevieDefferedMessage using queue", async function(): Promise<
```

Comment:
Change this to `Abandoning a deferred message returns it to deferred queue`
Same for subscription